### PR TITLE
Move cdap-cli and cdap-explore-jdbc out of SDK lib directory

### PR DESCRIPTION
**Pull request by chtyim**
- Those are uber jar, which affect correctness of program class loader construction

### CODE REVIEW

**Comment by chtyim** [comment]
There is a follow-up removal of hive-exec as well: https://issues.cask.co/browse/CDAP-4151

File: cdap-cli/bin/cdap-cli.sh [hunk 1]
```diff
@@ -34,14 +34,17 @@ done
 bin=`dirname "${PRG}"`
 bin=`cd "$bin"; pwd`
 lib="$bin"/../lib
+libexec="$bin"/../libexec
 conf="$bin"/../conf
 CDAP_CONF=${CDAP_CONF:-/etc/cdap/conf}
 script=`basename $0`
 
+CLI_CP=${libexec}/co.cask.cdap.cdap-cli-@@project.version@@.jar:${lib}/co.cask.cdap.cdap-cli-@@project.version@@.jar
```
Comment (anew) [inline cdap-cli/bin/cdap-cli.sh]: why keep the old path?

File: cdap-cli/bin/cdap-cli.bat [hunk 2]
```diff
@@ -22,7 +22,7 @@ SET CDAP_HOME=%~dp0
 SET CDAP_HOME=%CDAP_HOME:~0,-5%
 SET JAVACMD=%JAVA_HOME%\bin\java.exe
 
-SET CLASSPATH=%CDAP_HOME%\lib\co.cask.cdap.cdap-cli-@@project.version@@.jar;%CDAP_HOME%\conf
+SET CLASSPATH=%CDAP_HOME%\libexec\co.cask.cdap.cdap-cli-@@project.version@@.jar;%CDAP_HOME%\lib\co.cask.cdap.cdap-cli-@@project.version@@.jar;%CDAP_HOME%\conf
```
Comment (anew) [inline cdap-cli/bin/cdap-cli.bat]: why keep the old path?

File: cdap-cli/bin/cdap-cli.sh (hunk 1 above)
Comment (chtyim) [inline cdap-cli/bin/cdap-cli.sh]: The script is shared by SDK and RPM/DEB packaging. In distributed mode, those jars are still copied to "lib", but it's under `/opt/cdap/cli/lib`, which doesn't affect how master operates.

File: cdap-cli/bin/cdap-cli.bat (hunk 2 above)
Comment (chtyim) [inline cdap-cli/bin/cdap-cli.bat]: Same reason as above (or actually no one use .bat in distributed mode??).

File: cdap-standalone/pom.xml [hunk 3]
```diff
@@ -195,7 +210,25 @@
...
+                <phase>prepare-package</phase>
+                <goals>
+                  <goal>copy-dependencies</goal>
+                </goals>
+                <configuration combine.self="override">
+                  <outputDirectory>${stage.libexec.dir}</outputDirectory>
+                  <overWriteReleases>false</overWriteReleases>
+                  <overWriteSnapshots>false</overWriteSnapshots>
+                  <overWriteIfNewer>true</overWriteIfNewer>
+                  <includeArtifactIds>cdap-cli,cdap-explore-jdbc</includeArtifactIds>
```
Comment (anew) [inline cdap-standalone/pom.xml]: how does explore k now where to pick up the jar from libexec?

Comment (chtyim) [inline cdap-standalone/pom.xml]: Explore never uses explore-jdbc. It's only for distribution.

File: cdap-cli/bin/cdap-cli.sh (hunk 1 above)
Comment (anew) [inline cdap-cli/bin/cdap-cli.sh]: I see. Then it makes sense.

File: cdap-cli/bin/cdap-cli.bat (hunk 2 above)
Comment (anew) [inline cdap-cli/bin/cdap-cli.bat]: OK to keep it in parity with the .sh script

File: cdap-standalone/pom.xml (hunk 3 above)
Comment (anew) [inline cdap-standalone/pom.xml]: Then do we need to document the new location?

Comment (chtyim) [inline cdap-standalone/pom.xml]: Updated the doc.

**Comment by anew** [comment]
LGTM

**Comment by nitinmotgi** [comment]
I have tested this PR and it works. It also allows algorithmia-plugins to work without issues. LGTM. Merging it.

**Pull request merged**